packages/ui/src/components/vuestic-components/va-dropdown/VaDropdown.vue

```diff
@@ -128,15 +117,34 @@ export default class VaDropdown extends mixins(DropdownPropsMixin) {
     })
   }
 
+  handleClick (emitName: ClickType, toClose: boolean): void {
+    this.$emit(emitName)
+    if (toClose) {
+      this.hide()
+    }
+  }
+
+  onDropdownContentClick (): void {
+    this.handleClick('dropdown-content-click', this.closeOnClickInside)
+  }
+
+  onClickOutside (): void {
+    this.handleClick('click-outside', this.closeOnClickOutside)
+  }
+
   onAnchorClick (): void {
-    this.$emit('anchor-click')
     if (this.disabled) {
       return
     }
-    if (this.isClicked && !this.closeOnAnchorClick) {
-      return
+    if (this.trigger === 'click') {
+      if (this.valueComputed) {
+        this.handleClick('anchor-click', this.closeOnAnchorClick)
```

Comment:
Why is this event getting triggered only when dropdown is closed?
Wouldn't it make sense to trigger it on any anchor-click?

Reply:
We emit it when dropdown is opened as well - by "handleClick" method, not only when it's closed.

We do not emit only when it is disabled but if we need to emit in this case as well I can change it.